lib/galaxy/authnz/custos_authnz.py

```diff
@@ -106,7 +106,6 @@ def callback(self, state_token, authz_code, trans, login_redirect_url):
                         raise Exception("There already exists a user with email %s.  To associate this external login, you must first be logged in as that existing account." % email)
                 else:
                     user = trans.app.user_manager.create(email=email, username=username)
-                    user.set_random_password()
```

Comment:
@nuwang Does it require updating the following method as well?
https://github.com/galaxyproject/galaxy/blob/7855103d2c3846772c20299a06b43e94db8ed201/lib/galaxy/model/__init__.py#L5706-L5719

Reply:
@VJalili Thanks for pointing that out. I think that can remain as is, because if kwargs happens to contain a password, that will be ignored and always overwritten with a random password explicitly. So no change in semantics.